src/network/netdev/tunnel.c

```diff
@@ -76,6 +81,34 @@ static int netdev_ipip_fill_message_create(NetDev *netdev, Link *link, sd_netlin
         if (r < 0)
                 return log_netdev_error_errno(netdev, r, "Could not append IFLA_IPTUN_PMTUDISC attribute: %m");
 
+
+        if (t->fou_tunnel) {
+
+                switch (t->fou_encap_type) {
+                case NETDEV_FOO_OVER_UDP_ENCAP_DIRECT:
+                        t->fou_encap_type = FOU_ENCAP_DIRECT;
+                        break;
+                case NETDEV_FOO_OVER_UDP_ENCAP_GUE:
+                        t->fou_encap_type = FOU_ENCAP_GUE;
+                        break;
+                default:
+                        t->fou_encap_type = FOU_ENCAP_DIRECT;
```

Comment:
You're overwriting `fou_encap_type` with a different value. This looks wrong. Can you explain what is going on here?